library/n2_protocol.hpp

```diff
@@ -13,6 +13,11 @@ struct data_in_file {
 	off_t local_offset = 0;
 	size_t fsize = 0;
 	int on_exit = 0;
+
+	~data_in_file();
```

Comment:
Ok. At this time I can make private only `on_exit_`.

Reply:
It's a bad style. Please, use one of these patterns:
1) struct without methods (except trivial constructors) and private section.
2) class with all members as private.

Reply:
In my opinion, they're not the only variants, we can discuss in the chat. Public interface of class/struct depends on usage. If object is action-oriented, public interface is usually made from methods. If object is data oriented, public interface may have data.

For example, if we'll strictly be driven by suggested pattern, all requests/responses must became classes. But since they are data oriented, it'll lead only to overgrowth of code (writing trivial proxy accessors and etc.).

Yes, we can put all the public data into struct, and make one getter: `const data_of_the_class &the_class::data()`. Sometimes it's good pattern.

Reply:
> if we'll strictly be driven by suggested pattern, all requests/responses must became classes

I think yes, and we did concession for saving time for refactoring. But for all new code writing a code in good style is recommended.